microprofile-metrics/src/test/java/org/jboss/eap/qe/microprofile/metrics/MetricsTest.java

```diff
@@ -0,0 +1,320 @@
+package org.jboss.eap.qe.microprofile.metrics;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.hasKey;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.lessThanOrEqualTo;
+
+import java.net.URL;
+import java.util.Arrays;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.eap.qe.microprofile.metrics.hello.MetricsApp;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import io.restassured.http.ContentType;
+import io.restassured.specification.RequestSpecification;
+
+@RunWith(Arquillian.class)
+public class MetricsTest {
+
+    @ArquillianResource
+    URL deploymentUrl;
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    private RequestSpecification jsonMetricsRequest;
+    private RequestSpecification textMetricsRequest;
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return ShrinkWrap.create(WebArchive.class, MetricsTest.class.getSimpleName() + ".war")
+                .addPackage(MetricsApp.class.getPackage())
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml");
+    }
+
+    @Before
+    public void prepare() {
+        jsonMetricsRequest = given()
+                .baseUri("http://" + managementClient.getMgmtAddress())
+                .port(managementClient.getMgmtPort())
+                .basePath("metrics")
+                .accept(ContentType.JSON);
+        textMetricsRequest = given()
+                .baseUri("http://" + managementClient.getMgmtAddress())
+                .port(managementClient.getMgmtPort())
+                .basePath("/metrics")
+                .accept(ContentType.TEXT);
+    }
+
+    /**
+     * @tpTestDetails Smoke scenario to verify correct behaviour
```

Comment:
"Smoke" keyword. What is "correct behavior"?

Reply:
It is smoke test. Correct behavior of MP Metrics. See pass criteria.

Reply:
Please see email from @rsvoboda from Nov 25, 2019, 10:58 AM

Reply:
Something like: "Test to verify specification compatibility..." would be IMHO better.